src/AWS.Deploy.Orchestration/DeploymentBundleHandler.cs

```diff
@@ -108,21 +112,70 @@ public async Task PushDockerImageToECR(Recommendation recommendation, string rep
             recommendation.DeploymentBundle.ECRImageTag = tagSuffix;
         }
 
+        /// <summary>
+        /// The supported .NET versions on Elastic Beanstalk are dependent on the available platform versions.
+        /// These versions do not always have the required .NET runtimes installed so we need to perform extra checks
+        /// and perform a self-contained publish when creating the deployment bundle if needed.
+        /// </summary>
+        private void SwitchToSelfContainedBuildIfNeeded(Recommendation recommendation)
+        {
+            if (recommendation.Recipe.TargetService == RecipeIdentifier.TARGET_SERVICE_ELASTIC_BEANSTALK)
+            {
+                var targetFramework = recommendation.ProjectDefinition.TargetFramework ?? string.Empty;
+
+                // Elastic Beanstalk doesn't currently have .NET 7 preinstalled.
+                var unavailableFramework = new List<string> { "net7.0" };
+                var frameworkNames = new Dictionary<string, string> { { "net7.0", ".NET 7" } };
+                if (unavailableFramework.Contains(targetFramework))
+                {
+                    _interactiveService.LogInfoMessage($"Using self-contained publish since AWS Elastic Beanstalk does not currently have {frameworkNames[targetFramework]} preinstalled");
+                    recommendation.DeploymentBundle.DotnetPublishSelfContainedBuild = true;
+                    return;
+                }
+
+                var beanstalkPlatformSetting = recommendation.Recipe.OptionSettings.FirstOrDefault(x => x.Id.Equals("ElasticBeanstalkPlatformArn"));
+                if (beanstalkPlatformSetting != null)
+                {
+                    var beanstalkPlatformSettingValue = _optionSettingHandler.GetOptionSettingValue<string>(recommendation, beanstalkPlatformSetting);
+                    var beanstalkPlatformSettingValueSplit = beanstalkPlatformSettingValue?.Split("/");
+                    if (beanstalkPlatformSettingValueSplit?.Length != 3)
+                        throw new InvalidElasticBeanstalkPlatformException(DeployToolErrorCode.InvalidElasticBeanstalkPlatform, $"The selected Elastic Beanstalk platform version '{beanstalkPlatformSettingValue}' is invalid.");
+                    var beanstalkPlatformName = beanstalkPlatformSettingValueSplit[1];
+                    if (!Version.TryParse(beanstalkPlatformSettingValueSplit[2], out var beanstalkPlatformVersion))
+                        throw new InvalidElasticBeanstalkPlatformException(DeployToolErrorCode.InvalidElasticBeanstalkPlatform, $"The selected Elastic Beanstalk platform version '{beanstalkPlatformSettingValue}' is invalid.");
```

Comment:
Is this format part of Elastic Beanstalk's "contract"? I would hate to penalize users by throwing an exception if EB ever changed the convention of these strings. 

I'm wondering if it'd be safer to just proceed and possibly let the deployment fail, since users could then set `Self Contained Build` themselves.

Reply:
Done